syntax = "proto3";

package monitor;

// for testing purpose
service Monitor {
    rpc PersistHeartbeat (HeartBeat) returns (HeartBeatReceivedACK) {}
}

message HeartBeatWrapper {
    HeartBeat node_status_desc = 1;
}

message HeartBeat {
    string node_ip = 1;
    string node_status = 2; // "HEALTHY"
    string timestamp = 3;
    string response_time = 4;
}

message HeartBeatReceivedACK {
    string successACK = 1; //"Heartbeat received"
}